### DISCUSSION TITLE
Add support for a proprietary extension for template files

### DISCUSSION THREAD

**Discussion by ssssct** [ExpressionEngine/ExpressionEngine#4062] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
EE templates are using a proprietary template language which is definitely not valid HTML, so using `.html` as file extension for template files is a nonsense, and it is adding a lot of unnecessary complexity for syntax validators, code formatters, etc.

**Describe the solution you'd like**
EE template files could use a proprietary file extension (maybe something like `.ee` or `.eeml`) that would allow to identify the files for what they are. 

That new file extension would be defined by EE, not by the end user, and to keep retro compatibility with `.html` templates a config setting (boolean?) could be added so the end user could choose to keep using `.html`.

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
Can't say whether this is a good or bad idea... Definitely worth consideration, however one needs to remember that different template files would have different extensions currently - e.g. Stylesheet templates would have `.css` and Javascript templates would have `.js`

**Comment by ssssct** [comment]
If the Javascript and CSS templates contain EE template tags, they are not valid CSS or JS files either. So from my point of view, they should not be named `.css` or `.js`.
